tests/e2e/test_client.py

```diff
@@ -193,6 +197,38 @@ def test_upload_csv_success(self):
             True,
         )
 
+    def test_download_csv_success_by_id(self):
+        client = DuneClient(self.valid_api_key)
+        new_query = self.copy_query_and_change_parameters()
+        # Run query with new parameters
+        client.run_query(new_query)
+        # Download CSV by query_id
+        result_csv = client.download_csv(self.query.query_id)
+        # Expect that the csv returns the latest execution results (i.e. those that were just run)
+        self.assertEqual(
+            pandas.read_csv(result_csv.data).to_csv(index=False),
+            "text_field,number_field,date_field,list_field\n"
+            "different word,22,1991-01-01 00:00:00.000,Option 2\n",
+        )
+
+    def test_download_csv_success_with_params(self):
+        client = DuneClient(self.valid_api_key)
+        # Download CSV with query and given parameters.
+        result_csv = client.download_csv(self.query)
+        # Expect the result to be relative to values of given parameters.
+        #################################################################
+        # Note that we could compare results with
+        #    ",".join([p.value for p in self.query.parameters()]) + "\n"
+        # but there seems to be a discrepancy with the date string values.
+        # Specifically 1991-01-01 00:00:00.000
+        #           vs 1991-01-01 00:00:00
+        #################################################################
+        self.assertEqual(
+            pandas.read_csv(result_csv.data).to_csv(index=False),
+            "text_field,number_field,date_field,list_field\n"
```

Comment:
Dune's API interface doesn't guarantee that the returned csv will have a specific column ordering.
Not sure if Dune always return in a specific order today, but even if that is true today it maybe not be in the future.
Using to_json instead of to_csv will make tests a little bit less brittle here. 
WDYT?

Reply:
Very nice point! Will do.

Reply:
Fixed!